qml/ValueReadout.qml

```diff
@@ -119,13 +118,24 @@ Item {
         property alias readoutFontSize: readout.font.pointSize
     }
 
-
-    Component.onCompleted: {
-        if(y <= 0 || y + height > parent.height) {
-            y = margin
+    function checkPosition() {
+        // Check if parent is visible and correct initialized
+        if(!root.parent.visible || root.parent.width === 0 || root.parent.height === 0) {
+            return
+        }
+        if(y <= 0 || y + root.height > parent.height) {
+            y = root.parent.height - root.height - margin
         }
-        if(x <= 0 || x + width > parent.width) {
+        if(x <= 0 || x + root.width > parent.width) {
```

Comment:
How does this work?
Why is the x = margin on both upper and lower limits?
Wouldn't it make more sense to just limit to upper and lower bounds?
why not just `x = Math.max(Math.min(0, x), parent.height - root.height)`? 
Or even better, make a more readable `clamp(number, min, max)` based on that.

Reply:
> Why is the x = margin on both upper and lower limits?

This code is just to avoid an outbound position of readout.

> why not just x = Math.max(Math.min(0, x), parent.height - root.height)?

Yes, this is an alternative.